### DISCUSSION TITLE
Preference for merging PRs

### DISCUSSION THREAD

**Discussion by Jint-lzxy** [ayamir/nvimdots#726] [General]
So the way PRs are merged now is not the same. Basically we have three options: "Create a merge commit", "Squash and merge" and "Rebase and merge". Both @CharlesChiuGit and I chose "Squash and merge" and @ayamir chose "Create a merge commit".

FWIW we should unify the way of merging PRs. I tend to use "Squash and merge" b/c it can maintain linear commit history & make the commit tree more concise. What do you guys think?

<img width="300" alt="info" src="https://user-images.githubusercontent.com/50296129/236841743-621e3e01-db5b-4498-9f1b-b56847d83385.png">

**Comment by charliie-dev** [comment]
second with u, and the reasons are the same.

**Comment by ayamir** [comment]
>  I tend to use "Squash and merge" b/c it can maintain linear commit history & make the commit tree more concise.

ok, I use the former just because it's the default option.
